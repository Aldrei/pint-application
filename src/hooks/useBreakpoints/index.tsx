import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

export const useBreakpoints = () => {
  const theme = useTheme();

  const goSm = useMediaQuery(theme.breakpoints.up('sm'));
  const goMd = useMediaQuery(theme.breakpoints.up('md'));
  const goLg = useMediaQuery(theme.breakpoints.up('lg'));
  const goXl = useMediaQuery(theme.breakpoints.up('xl'));

  return [goSm, goMd, goLg, goXl];
};
