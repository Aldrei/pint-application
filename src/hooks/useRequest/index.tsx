import * as React from 'react';

import axios from '../../hooks/useConfigAxios';

// import { propertiesVideosService } from '../../services/properties';

import { IServiceRequestStatus } from '../../types';

interface IProps {
  type: 'propertyVideoDelete';
  code: string;
  videoId: string;
}

const useRequest = ({ code, videoId }: IProps) => {
  const [status, setStatus] = React.useState<IServiceRequestStatus>('idle');

  axios.get('https://swapi.dev/api/people', {
    onDownloadProgress: progressEvent => {
      setStatus('loading');
    }
  })
    .then((res) => {
      console.log('DEBUG RESPONSE:', res);
      setStatus('success');
    })
    .catch((err) => {
      console.log('DEBUG ERROR:', err);
      setStatus('failed');
    });

  return { status };
};

export default useRequest;