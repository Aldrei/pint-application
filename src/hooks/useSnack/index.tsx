import React from 'react';

import Snackbar from '@mui/material/Snackbar';
import MuiAlert, { AlertProps } from '@mui/material/Alert';

const Alert = React.forwardRef<HTMLDivElement, AlertProps>(function Alert(props, ref) {
  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
});

const useSnack = () => {
  const [snacks, setSnacks] = React.useState(false);

  const handleClose = () => setSnacks(false);

  const SnackList = React.useCallback(() => {
    return (
      <Snackbar 
        open={snacks} 
        autoHideDuration={6000} 
        onClose={handleClose}
        sx={{ '& .MuiPaper-root': { flexDirection: 'row' } }}
      >
        <Alert 
          onClose={handleClose} 
          severity="success" 
          sx={{ width: '100%' }}
        >
          This is a success message!
        </Alert>
      </Snackbar>
    );
  }, [snacks]);

  const addSnack = React.useMemo(() => ({
    addMessage: () => {
      console.log('ADD SNACK MESSAGE!');
      setSnacks(!snacks);
    },
  }), []);

  return { addSnack, snacks, SnackList };
};

export default useSnack;