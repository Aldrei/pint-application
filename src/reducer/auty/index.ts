import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse } from '../../services/auth';

import { IServiceRequest } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: object;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  status: 'idle',
  whoIsAuth: {},
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

// The function below is called a thunk and allows us to perform async logic. It
// can be dispatched like a regular action: `dispatch(incrementAsync(10))`. This
// will call the thunk with the `dispatch` function as the first argument. Async
// code can then be executed and other actions can be dispatched. Thunks are
// typically used to make async requests.
export const authServiceThunk = createAsyncThunk(
  'counter/fetchCount',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload
    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {},
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        console.log('authServiceThunk fulfilled:', action);
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        console.log('authServiceThunk rejected:', action);
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      });
  },
});

// The function below is called a selector and allows us to select a value from
// the state. Selectors can also be defined inline where they're used instead of
// in the slice file. For example: `useSelector((state: RootState) => state.counter.value)`
// export const selectCount = (state: RootState) => state.counter.value;

export default authSlice.reducer;
