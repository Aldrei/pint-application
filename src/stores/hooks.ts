import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './index';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useSelectorCallback = (state: RootState, reducer: keyof RootState) => state[reducer];
