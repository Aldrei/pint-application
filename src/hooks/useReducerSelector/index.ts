import { RootState, IReducersType } from '../../store';
import { useAppSelector } from '../../store/hooks';

export const useAppSelectorBlaBlaBal = (reducer: keyof IReducersType) => useAppSelector((state: RootState) => state[reducer]);