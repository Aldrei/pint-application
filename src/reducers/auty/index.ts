import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse, IAuthServiceRevokeRequest } from '../../services/auth';
import { AppThunk } from '../../stores';

import { IServiceRequest, IWhoIsAuth } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: IWhoIsAuth;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  name: 'authReducer',
  status: 'idle',
  whoIsAuth: {} as IWhoIsAuth,
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

export const authServiceThunk = createAsyncThunk(
  'auth/access_token',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const revokeServiceThunk = createAsyncThunk(
  'auth/revoke',
  async (data: IAuthServiceRevokeRequest) => {
    const response = await authService.revoke(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const whoServiceThunk = createAsyncThunk(
  'auth/who_is_auth',
  async () => {
    const response = await authService.who();
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    clearAccessToken: (state) => {
      state.accessToken = {} as IAuthServiceAccessTokenResponse;
      state.whoIsAuth = {} as IWhoIsAuth;
    }
  },
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      /** Access token */
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      })
      /** Revoke. */
      .addCase(revokeServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(revokeServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.data = action.payload.data;
        state.accessToken = {};
        state.whoIsAuth = {} as IWhoIsAuth;
      })
      .addCase(revokeServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
      })
      /** Who is auth. */
      .addCase(whoServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(whoServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.whoIsAuth = action.payload.data;
      })
      .addCase(whoServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.whoIsAuth = action.payload as IWhoIsAuth;
      });
  },
});

const { clearAccessToken } = authSlice.actions;

export const clearAccessTokenReducer = (): AppThunk => (dispatch) => {
  dispatch(clearAccessToken());
};

export default authSlice.reducer;
