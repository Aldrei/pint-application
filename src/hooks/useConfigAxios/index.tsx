import axios from 'axios';

import { clearAccessTokenReducer } from '../../reducers/auty';
import { useAppDispatch } from '../../stores/hooks';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_BASE_URL,
});

const getToken = () => {
  try {
    const persistImob = localStorage.getItem('persist:imob');
    if (persistImob) {
      const persistImobData = JSON.parse(persistImob);

      if (persistImobData.authReducer) {
        const authReducerData = Object.assign({}, JSON.parse(persistImobData.authReducer));
        return authReducerData?.accessToken?.access_token;
      }
    }

    return '';
  } catch (error) {
    console.log(error);
    return '';
  }
};


// Add a request interceptor
api.interceptors.request.use(function (config) {
  if (config.headers) config.headers['Authorization'] = `Bearer ${getToken()}`;
  return config;
}, function (error) {
  // Do something with request error
  return Promise.reject(error);
});

export const useConfigAxios = () => {
  const dispatch = useAppDispatch();

  // Add a response interceptor
  api.interceptors.response.use((response) => {
    console.log('api.interceptors.response success...');
    // Any status code that lie within the range of 2xx cause this function to trigger
    /** NOTE: Suppress 'A non-serializable value was detected in the state, in the path' error */
    return JSON.parse(JSON.stringify(response));
  }, (error) => {
    console.log('api.interceptors.response error...', { error });
    // Any status codes that falls outside the range of 2xx cause this function to trigger
    // return Promise.reject(error);

    if (error.response.status === 401) {
      dispatch(clearAccessTokenReducer());
    }

    /** NOTE: Suppress 'A non-serializable value was detected in the state, in the path' error */
    return JSON.parse(JSON.stringify(error.response));
  });

  return [];
};

export default api;