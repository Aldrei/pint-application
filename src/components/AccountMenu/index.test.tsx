import { fireEvent, render, waitFor } from '@testing-library/react';
import React from 'react';

import { useAppSelectorBlaBlaBal } from '../../hooks/useReducerSelector';
import { useAppDispatch } from '../../hooks/useReducerDispatch';
import { IAutyState } from '../../reducers/auty';

import AccountMenu from './index';

jest.mock('../../hooks/useReducerSelector', () => ({
  useAppSelectorBlaBlaBal: jest.fn(),
}));

jest.mock('../../hooks/useReducerDispatch', () => ({
  useAppDispatch: jest.fn(),
}));

describe('AccountMenu component', () => {
  const useAppSelectorBlaBlaBalMocked = useAppSelectorBlaBlaBal as jest.MockedFunction<typeof useAppSelectorBlaBlaBal>;
  const useAppDispatchMocked = useAppDispatch as jest.MockedFunction<typeof useAppDispatch>;

  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Should render closed menu', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);
    // useAppDispatchMocked.mockReturnValue();

    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const nodeEl = render(<AccountMenu />);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should render opened menu', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);

    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const nodeEl = render(<AccountMenu />);

    const accountMenuButton = nodeEl.getByTestId('account-menu-button-open-testid');
    fireEvent.click(accountMenuButton);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should close menu', async () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);

    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const nodeEl = render(<AccountMenu />);

    expect(nodeEl.queryByTestId('account-menu-list-testid')).not.toBeInTheDocument();

    expect(nodeEl.baseElement).toMatchSnapshot();


    const accountMenuButton = nodeEl.getByTestId('account-menu-button-open-testid');
    fireEvent.click(accountMenuButton);
    expect(nodeEl.queryByTestId('account-menu-list-testid')).toBeInTheDocument();

    expect(nodeEl.baseElement).toMatchSnapshot();


    const accountMenuList = nodeEl.getByTestId('account-menu-list-testid');
    fireEvent.click(accountMenuList);

    await waitFor(() => {
      expect(nodeEl.queryByTestId('account-menu-list-testid')).not.toBeInTheDocument();
    });

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should click logout option', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);
    useAppDispatchMocked.mockReturnValue(jest.fn());

    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const nodeEl = render(<AccountMenu />);

    const accountMenuButton = nodeEl.getByTestId('account-menu-button-open-testid');
    fireEvent.click(accountMenuButton);

    const accountMenuLogoutButton = nodeEl.getByTestId('account-menu-list-logout-option');
    fireEvent.click(accountMenuLogoutButton);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });
});
