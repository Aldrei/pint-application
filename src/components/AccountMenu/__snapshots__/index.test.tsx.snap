// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountMenu component Should click logout option 1`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root MuiMenu-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
    data-testid="account-menu-list-testid"
    id="account-menu"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiPopover-paper css-wnyqvh-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
      style="opacity: 0; transform: scale(0.75, 0.5625); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root Mui-focusVisible css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="account-menu-list-logout-option"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
          </div>
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AccountMenu component Should close menu 1`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`AccountMenu component Should close menu 2`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-controls="account-menu"
        aria-expanded="true"
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPopover-root MuiMenu-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
    data-testid="account-menu-list-testid"
    id="account-menu"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiPopover-paper css-wnyqvh-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root Mui-focusVisible css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="account-menu-list-logout-option"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
          </div>
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AccountMenu component Should close menu 3`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`AccountMenu component Should render closed menu 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`AccountMenu component Should render opened menu 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="MuiBox-root css-5nwj3y"
    >
      <button
        aria-controls="account-menu"
        aria-expanded="true"
        aria-haspopup="true"
        aria-label="Perfil"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="account-menu-button-open-testid"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
        >
          M
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPopover-root MuiMenu-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
    data-testid="account-menu-list-testid"
    id="account-menu"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiPopover-paper css-wnyqvh-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root Mui-focusVisible css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="account-menu-list-logout-option"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
              />
            </svg>
          </div>
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
